File: src/com/owlrobotics/pedropathingvisualizer/pedropathing/util/SingleRunAction.java
package com.owlrobotics.pedropathingvisualizer.pedropathing.util;

public class SingleRunAction {
    private boolean hasBeenRun;
    private Runnable runnable;

    public SingleRunAction(Runnable runnable) {
        this.runnable = runnable;
    }

    public boolean hasBeenRun() {
        return this.hasBeenRun;
    }

    public boolean run() {
        if (!this.hasBeenRun) {
            this.hasBeenRun = true;
            this.runnable.run();
            return true;
        } else {
            return false;
        }
    }

    public void reset() {
        this.hasBeenRun = false;
    }
}

File: src/com/owlrobotics/pedropathingvisualizer/pedropathing/util/SingleRunAction.kt
package com.owlrobotics.pedropathingvisualizer.pedropathing.util

open class SingleRunAction(private val runnable: Runnable) {
    private var hasBeenRun = false

    fun hasBeenRun(): Boolean {
        return this.hasBeenRun
    }

    fun run(): Boolean {
        if (!this.hasBeenRun) {
            this.hasBeenRun = true
            runnable.run()
            return true
        } else {
            return false
        }
    }

    fun reset() {
        this.hasBeenRun = false
    }
}
